eZ/Publish/Core/Limitation/ObjectStateLimitationType.php

```diff
@@ -123,11 +123,11 @@ public function evaluate(APILimitationValue $value, APIUserReference $currentUse
         } elseif ($object instanceof VersionInfo) {
             $object = $object->getContentInfo();
         } elseif (!$object instanceof ContentInfo && !$object instanceof ContentCreateStruct) {
-            throw new InvalidArgumentException('$object', 'Must be of type: Content, VersionInfo or ContentInfo');
+            throw new InvalidArgumentException('$object', 'Must be of type: Content, VersionInfo, ContentInfo or ContentCreateStruct');
         }
 
         // Skip evaluating for RootLocation
-        if (1 === $object->mainLocationId) {
+        if (isset($object->mainLocationId) && 1 === $object->mainLocationId) {
```

Comment:
The solution here is to check for main location only if $object is a specific instance. Which instance contains the given property is a well known fact. It will become easier to refactor in the future (planning to introduce strategy per object type to make it more SOLID.